### PR TITLE
🎓 Testing C to F Tests and Automation

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
- Include another example of testing (`celsius_to_fahrenheit`)
- Mention `unittest`, but that's it. 

### Why
The more the merrier.

### CODE REVIEW

File: site/topics/testing/testing.rst [hunk 1]
```diff
@@ -165,12 +165,57 @@ Square of Sums Example Tests
...
+
+
+* Above is a series of assertion tests for the ``celsius_to_fahrenheit`` function
+* Notice the key, broad tests
+
+    * Input of zero
+    * Negative input
+    * Positive input
+    * Input is a float
+    * Output is a float
```
Comment (twentylemon) [inline site/topics/testing/testing.rst:200]: I'd argue `-40` is a special case and does not cover the general negative number case. It's the only point where the two scales are equal, which sounds like an edge case to me.

_Test driven development_ says you actually write some failing tests first, before writing the code that makes the tests pass. Once the tests all pass, either you add more failing tests and iterate to get the full behaviour you want, or you're done. I'm not one to say anyone should follow TDD (I don't enjoy inflicting pain on others), but there is a benefit to at least thinking about tests _first_. Thinking about the problem you're solving before there's a solution, I'll just assert, is very helpful.

Not really applicable here, but I also say that tests are the first clients of your code. If you find something hard to test, it's likely hard to use.

Comment (jameshughes89) [reply site/topics/testing/testing.rst:200]: > Not really applicable here, but I also say that tests are the first clients of your code. If you find something hard to test, it's likely hard to use.

image

**Review by twentylemon — APPROVED** [review]

**Review by twentylemon — APPROVED** [review]
